controller/api/public/grpc_server.go

```diff
@@ -63,19 +67,27 @@ func newGrpcServer(
 ) *grpcServer {
 
 	grpcServer := &grpcServer{
-		prometheusAPI:       promAPI,
-		tapClient:           tapClient,
-		discoveryClient:     discoveryClient,
-		k8sAPI:              k8sAPI,
-		controllerNamespace: controllerNamespace,
-		ignoredNamespaces:   ignoredNamespaces,
+		prometheusAPI:         promAPI,
+		tapClient:             tapClient,
+		discoveryClient:       discoveryClient,
+		k8sAPI:                k8sAPI,
+		controllerNamespace:   controllerNamespace,
+		ignoredNamespaces:     ignoredNamespaces,
+		mountPathGlobalConfig: pkgK8s.MountPathGlobalConfig,
+		mountPathProxyConfig:  pkgK8s.MountPathProxyConfig,
 	}
 
 	pb.RegisterApiServer(prometheus.NewGrpcServer(), grpcServer)
 
 	return grpcServer
 }
 
+func (s *grpcServer) WithConfigPaths(global, proxy string) *grpcServer {
```

Comment:
Since this is only used in tests, my preference would be to remove this method so that people can't misuse it, and instead just set these fields directly in the test. Something like:

```diff
diff --git a/controller/api/public/grpc_server.go b/controller/api/public/grpc_server.go
index 43ff9edb..d8b0a121 100644
--- a/controller/api/public/grpc_server.go
+++ b/controller/api/public/grpc_server.go
@@ -82,12 +82,6 @@ func newGrpcServer(
 	return grpcServer
 }
 
-func (s *grpcServer) WithConfigPaths(global, proxy string) *grpcServer {
-	s.mountPathGlobalConfig = global
-	s.mountPathProxyConfig = proxy
-	return s
-}
-
 func (*grpcServer) Version(ctx context.Context, req *pb.Empty) (*pb.VersionInfo, error) {
 	return &pb.VersionInfo{GoVersion: runtime.Version(), ReleaseVersion: version.Version, BuildDate: "1970-01-01T00:00:00Z"}, nil
 }
diff --git a/controller/api/public/grpc_server_test.go b/controller/api/public/grpc_server_test.go
index 95124a9f..4ebf424b 100644
--- a/controller/api/public/grpc_server_test.go
+++ b/controller/api/public/grpc_server_test.go
@@ -586,7 +586,9 @@ func TestConfig(t *testing.T) {
 			k8sAPI,
 			"linkerd",
 			[]string{},
-		).WithConfigPaths("testdata/global.conf.json", "testdata/proxy.conf.json")
+		)
+		fakeGrpcServer.mountPathGlobalConfig = "testdata/global.conf.json"
+		fakeGrpcServer.mountPathProxyConfig = "testdata/proxy.conf.json"
 
 		k8sAPI.Sync()
 
```

Reply:
Makes sense :+1: